### PR TITLE
GUI: Ask for a missing passphrase during automated installations

### PR DESCRIPTION
**Pull request by poncovka**
If the kickstart file defines a partitioning that requires a passphrase,
show a dialog that will allow users to provide the missing passphrase.

Resolves: rhbz#2029101

### CODE REVIEW

**Comment by poncovka** [comment]
What users see after starting an automated installation that requires a passphrase:

gui_require_passphrase

The installation automatically continues after the passphrase is provided.

**Comment by poncovka** [comment]
@jkonecny12, @jstodola, what do you think?

**Comment by jstodola** [comment]
:+1: Perfection

**Review by M4rtinK — APPROVED** [review]
Looks good to me. :)

Just a suggestion to improve a docstring.

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

File: pyanaconda/ui/gui/spokes/storage.py [hunk 1]
```diff
@@ -660,7 +660,12 @@ def _check_required_passphrase(self):
             return
 
         dialog = PassphraseDialog(self.data)
-        rc = dialog.run()
+
+        # Use MainWindow.get() instead of self.main_window,
+        # because the main_window property returns SpokeWindow
+        # instead of MainWindow during the initialization.
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/storage.py:666]: I'm missing reason why we need that.

**Review by jkonecny12 — APPROVED** [review]
Looks great to me. Thanks!

**Comment by jkonecny12** [comment]
Have all the BZ ACKs now.

File: pyanaconda/ui/gui/spokes/storage.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/ui/gui/spokes/storage.py:666]: Fixed.

**Comment by poncovka** [comment]
Rebased.

**Pull request merged**